### PR TITLE
feat(api): Standardize mechanism for ResourceHandlers to notify artifact deployments

### PR DESCRIPTION
**Pull request by luispollo**
This PR introduces two new methods to `ResourceHandler`, `notifyArtifactDeploying` and `notifyArtifactDeployed` which standardize the implementation of notifying core Keel of resource handlers beginning and successfully completing artifact deployments, allowing for external plugins to leverage the same mechanism.

### CODE REVIEW

**Comment by emjburns** [comment]
😍  I like this implementation so much better! Just confirming - things are wired up correctly for the existing titus and ec2 plugins, right? Just want to make sure you've tested that the events still fire when running locally

**Comment by luispollo** [comment]
> 😍 I like this implementation so much better! Just confirming - things are wired up correctly for the existing titus and ec2 plugins, right? Just want to make sure you've tested that the events still fire when running locally

Yup! Plus, I didn't touch any part of the tests that confirm the behavior around those events, so if the tests pass it means the events are fired! 🙂

File: keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceActuator.kt [hunk 1]
```diff
@@ -200,7 +200,7 @@ class ResourceActuator(
           }
         }
       } catch (e: ResourceCurrentlyUnresolvable) {
-        log.warn("Resource check for {} failed (hopefully temporarily) due to {}", id, e.message, e)
+        log.warn("Resource check for {} failed (hopefully temporarily) due to {}", id, e.message)
```
Comment (luispollo) [inline keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceActuator.kt:203]: This was causing stack traces to be printed for a case that's an expected condition. It was just noise in the logs.

File: keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/plugins/ResourceHandler.kt [hunk 2]
```diff
@@ -137,6 +153,26 @@ interface ResourceHandler<S : ResourceSpec, R : Any> : SpinnakerExtensionPoint {
...
   suspend fun actuationInProgress(resource: Resource<S>): Boolean = false
+
+  /**
+   * Notifies core Keel that a new artifact version is being deployed to the specified resource.
+   *
+   * The default implementation achieves this by publishing an [ArtifactVersionDeploying] event via the
+   * [EventPublisher], and should *not* be overridden by plugin implementations.
+   */
+  @JvmDefault
+  fun notifyArtifactDeploying(resource: Resource<S>, artifactVersion: String) =
```
Comment (lorin) [inline keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/plugins/ResourceHandler.kt:164]: I recommend marking these new methods as `protected` unless you foresee a scenario where they'll be called by anyone other than subclasses.

**Comment by emjburns** [comment]
LGTM!

**Comment by emjburns** [comment]
LGTM!

File: keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/plugins/ResourceHandler.kt (hunk 2 above)
Comment (luispollo) [reply keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/plugins/ResourceHandler.kt:164]: Can't because it's an interface...

**Pull request merged**